packages/protocol/test-sol/devchain/e2e/common/EpochManager.t.sol

```diff
@@ -677,6 +679,205 @@ contract E2E_EpochManager_FinishNextEpochProcess is E2E_EpochManager {
+    accounts.createAccount();
+    lockedCelo.lock{ value: ctx.lockedAmount }();
+    vm.stopPrank();
+
+    uint256 actualLocked = lockedCelo.getAccountTotalLockedGold(ctx.alice);
+    require(actualLocked == ctx.lockedAmount, "Alice lock failed");
+
+    (, GroupWithVotes[] memory initialGroups) = getGroupsWithVotes();
+    require(initialGroups.length >= 3, "Not enough groups for test setup");
+    ctx.targetGroup = initialGroups[2].group;
+  }
+
+  function _aliceVote(AliceContext memory ctx) internal {
+    (address lesser, address greater) = _calculateVoteNeighbors(ctx.targetGroup, ctx.lockedAmount);
+    vm.prank(ctx.alice);
+    election.vote(ctx.targetGroup, ctx.lockedAmount, lesser, greater);
+    vm.stopPrank();
+  }
+
+  function _aliceActivate(AliceContext memory ctx) internal {
+    vm.startPrank(ctx.alice);
+    election.activate(ctx.targetGroup);
+    vm.stopPrank();
+    assertAlmostEqual(election.getActiveVotesForGroupByAccount(ctx.targetGroup, ctx.alice), ctx.lockedAmount, 10, "Alice activation failed");
+  }
+
+   function _aliceRevoke(AliceContext memory ctx) internal {
+    (address lesser, address greater, uint index) = _calculateRevokeNeighbors(ctx.targetGroup, ctx.lockedAmount);
+
+    vm.startPrank(ctx.alice);
+    uint256 activeVotes = election.getTotalVotesForGroupByAccount(ctx.targetGroup, ctx.alice);
+    require(activeVotes >= ctx.lockedAmount, "Insufficient active votes to revoke");
+    election.revokeActive(ctx.targetGroup, ctx.lockedAmount, lesser, greater, index);
+    vm.stopPrank();
+  }
+
+  function _aliceUnlock(AliceContext memory ctx) internal {
+     uint256 unlockingPeriod = lockedCelo.unlockingPeriod();
+     timeTravel(unlockingPeriod + 1); 
+
+     vm.prank(ctx.alice);
+     lockedCelo.unlock(ctx.lockedAmount);
+     vm.stopPrank();
+  }
+
+  function _advanceAndFinishNextEpochProcessing(uint256 expectedStartEpoch) internal returns (uint256 newEpoch) {
+      assertEq(epochManagerContract.getCurrentEpochNumber(), expectedStartEpoch, "Epoch mismatch before finish");
+      timeTravel(epochDuration / 2);
+      blockTravel(100);
+
+      address[] memory lessers;
+      address[] memory greaters;
+      (lessers, greaters, ) = getLessersAndGreaters(groups); 
+
+      epochManagerContract.finishNextEpochProcess(groups, lessers, greaters);
+      newEpoch = epochManagerContract.getCurrentEpochNumber();
+      assertEq(newEpoch, expectedStartEpoch + 1, "Epoch did not increment after finish");
+  }
+
+   function _advanceAndStartNextEpochProcessing(uint256 expectedStartEpoch) internal {
+       assertEq(epochManagerContract.getCurrentEpochNumber(), expectedStartEpoch, "Epoch mismatch before start");
+       timeTravel(epochDuration + 1); 
+       epochManagerContract.startNextEpochProcess();
+       uint256 currentEpoch = epochManagerContract.getCurrentEpochNumber();
+       assertEq(currentEpoch, expectedStartEpoch, "Epoch number changed unexpectedly on startNextEpochProcess");
```

Comment:
Might be good for consistency to have the assersions in the actual test function instead of the helper functions.

Reply:
Unfortunately it is not possible at least for all of them - then we get error Stack too deep